board/ipaccess/ipa9131/characterisation.c

```diff
@@ -117,39 +143,91 @@ void serialise_characterisation_info(const struct characterisation_data_t * cd,
     payload[15] = (((cd->pcbai & 0xFF00) >> 8) & 0xFF);
     payload[16] = (cd->pcbai & 0xFF);
 
-    if (cd->version == 0x00)
-    {
-        payload[17] = cd->oui[0];
-        payload[18] = cd->oui[1];
-        payload[19] = cd->oui[2];
-        payload[20] = (((cd->serial & 0xFF000000) >> 24) & 0xFF);
-        payload[21] = (((cd->serial & 0x00FF0000) >> 16) & 0xFF);
-        payload[22] = (((cd->serial & 0x0000FF00) >>  8) & 0xFF);
-        payload[23] = (((cd->serial & 0x000000FF) >>  0) & 0xFF);
+    /*Calc md5sum*/
```

Comment:
U-Boot is really a bit brain dead in not providing MD5Init/MD5Update/MD5Final... such a pity, as it would have been much easier for you to use those had they been exposed.